File: backend/src/main/java/br/com/surb/mysales/dto/product/ProductResponse.java
package br.com.surb.mysales.dto.product;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.Instant;

public record ProductResponse(
   Long id,
   String sku,
   String name,
   BigDecimal price,
   String description,
   Instant createdAt
) implements Serializable {

  @Serial
  private static final long serialVersionUID = 5102161462547613039L;
}

File: frontend/src/http/response/ProductResponse.ts
export type ProductResponse = {
  id: number;
  sku: string;
  name: string;
  price?: number;
  description: string;
  createdAt?: string;
};

export type ProductFiledResponse = {
  id?: number;
  sku: string;
  name: string;
  price?: string;
  description: string;
  createdAt?: string;
};

export type ProductContent = {
  id: number;
  sku: string;
  name: string;
  price?: number;
};

export type ProductDTO = {
  id: number;
  sku: string;
  name: string;
  price: number;
  description: string;
  createdAt: string;
};

type ProductFormDTO = Omit<ProductDTO, 'id' | 'createdAt'>;
